shm/CMakeLists.txt

```diff
@@ -0,0 +1,29 @@
+cmake_minimum_required(VERSION 3.10)
+project(shm)
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED on)
+set(CMAKE_BUILD_TYPE Debug)
+
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/lib)
+set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${CMAKE_BINARY_DIR}/bin)
+
+set(INCLUDE_DIR ${CMAKE_SOURCE_DIR}/..)
+set(SOURCE_DIR ${CMAKE_SOURCE_DIR}/src)
+
+set(SOURCES
+    ${SOURCE_DIR}/Cargo.cpp
+    ${SOURCE_DIR}/Island.cpp
+    ${SOURCE_DIR}/Map.cpp
+    ${SOURCE_DIR}/Ship.cpp
```

Comment:
shouldn't we add main.cpp ?

Reply:
No, main.cpp is in 'add_executable()' section and it is linked with all the rest sources in 'target_link_libraries()' section.